native/membrane_videocompositor_wgpu/src/compositor/textures.rs

```diff
@@ -0,0 +1,244 @@
+#[repr(usize)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum YUVPlane {
+    Y = 0,
+    U,
+    V,
+}
+
+pub struct Texture {
+    desc: wgpu::TextureDescriptor<'static>,
+    texture: wgpu::Texture,
+    pub view: wgpu::TextureView,
+    pub bind_group: Option<wgpu::BindGroup>,
+}
+
+impl Texture {
+    fn new(
+        device: &wgpu::Device,
+        width: u32,
+        height: u32,
```

Comment:
Shouldn't this be grouped in the RawVideo struct?

Reply:
I don't really think it should, since this `Texture` struct is intended to be more general in use -- not necessarily associated with an input or output video. I also thought of the `RawVideo` struct as a purely rust-elixir interface related concept.